### DISCUSSION TITLE
could not compile `common-functions`

### DISCUSSION THREAD

**Discussion by chinawch007** [databendlabs/databend#4117] [Q&A] [answered]
When i make cli-build，i always meet “could not compile `common-functions`” error，therefore failed to complete compiling。
here is the output：

`error: could not compile common-functions
Caused by:
  process didn't exit successfully: rustc --crate-name common_functions --edition=2021 common/functions/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C embed-bitcode=no -C codegen-units=1 -C debuginfo=1 -C metadata=69c114325fc14244 -C extra-filename=-69c114325fc14244 --out-dir /home/thunderwang/git/databend/target/release/deps -L dependency=/home/thunderwang/git/databend/target/release/deps --extern base64=/home/thunderwang/git/databend/target/release/deps/libbase64-7d535c02f423e425.rmeta --extern blake3=/home/thunderwang/git/databend/target/release/deps/libblake3-a907fb8ebbeaedc3.rmeta --extern bstr=/home/thunderwang/git/databend/target/release/deps/libbstr-646d178dcaa83040.rmeta --extern bumpalo=/home/thunderwang/git/databend/target/release/deps/libbumpalo-d2ba47b8bb0e1c81.rmeta --extern bytes=/home/thunderwang/git/databend/target/release/deps/libbytes-850b925872cff88c.rmeta --extern common_arrow=/home/thunderwang/git/databend/target/release/deps/libcommon_arrow-767261a53612cb2c.rmeta --extern common_datavalues=/home/thunderwang/git/databend/target/release/deps/libcommon_datavalues-078614e2e53e83ff.rmeta --extern common_datavalues2=/home/thunderwang/git/databend/target/release/deps/libcommon_datavalues2-380e153955532bda.rmeta --extern common_exception=/home/thunderwang/git/databend/target/release/deps/libcommon_exception-ddafff1deeb1a11e.rmeta --extern common_io=/home/thunderwang/git/databend/target/release/deps/libcommon_io-62c225075724a9ee.rmeta --extern crc32fast=/home/thunderwang/git/databend/target/release/deps/libcrc32fast-839940f64c55efd2.rmeta --extern dyn_clone=/home/thunderwang/git/databend/target/release/deps/libdyn_clone-3651e5f6cb405887.rmeta --extern hex=/home/thunderwang/git/databend/target/release/deps/libhex-5e35002db5febb10.rmeta --extern indexmap=/home/thunderwang/git/databend/target/release/deps/libindexmap-aa6323733ecd0f38.rmeta --extern itertools=/home/thunderwang/git/databend/target/release/deps/libitertools-22370d3eb7223e11.rmeta --extern md5=/home/thunderwang/git/databend/target/release/deps/libmd5-ec9db1ceb814bc3c.rmeta --extern num=/home/thunderwang/git/databend/target/release/deps/libnum-44483e59cdfb58ed.rmeta --extern num_format=/home/thunderwang/git/databend/target/release/deps/libnum_format-dc837743861f064f.rmeta --extern num_traits=/home/thunderwang/git/databend/target/release/deps/libnum_traits-830efba7ded7aae1.rmeta --extern once_cell=/home/thunderwang/git/databend/target/release/deps/libonce_cell-16727d14885b4030.rmeta --extern ordered_float=/home/thunderwang/git/databend/target/release/deps/libordered_float-16a06a90908511c8.rmeta --extern paste=/home/thunderwang/git/databend/target/release/deps/libpaste-575099a02cca6656.so --extern rand=/home/thunderwang/git/databend/target/release/deps/librand-26f924503360aa29.rmeta --extern regex=/home/thunderwang/git/databend/target/release/deps/libregex-0dcba45aaca905b8.rmeta --extern serde=/home/thunderwang/git/databend/target/release/deps/libserde-774c793642f2cf29.rmeta --extern serde_json=/home/thunderwang/git/databend/target/release/deps/libserde_json-4462bfd64bb19827.rmeta --extern sha1=/home/thunderwang/git/databend/target/release/deps/libsha1-b629826246e685c8.rmeta --extern sha2=/home/thunderwang/git/databend/target/release/deps/libsha2-bc7f439bbd4621d1.rmeta --extern simdutf8=/home/thunderwang/git/databend/target/release/deps/libsimdutf8-ab7b473a6bc1ef91.rmeta --extern sqlparser=/home/thunderwang/git/databend/target/release/deps/libsqlparser-7919aca392590acf.rmeta --extern strength_reduce=/home/thunderwang/git/databend/target/release/deps/libstrength_reduce-b3a1c6143ae469e8.rmeta --extern twox_hash=/home/thunderwang/git/databend/target/release/deps/libtwox_hash-5265488753354cd1.rmeta --extern uuid=/home/thunderwang/git/databend/target/release/deps/libuuid-30cb8179c99a9f4c.rmeta -L native=/home/thunderwang/git/databend/target/release/build/blake3-da18a583f70b036d/out -L native=/home/thunderwang/git/databend/target/release/build/blake3-da18a583f70b036d/out -L native=/home/thunderwang/git/databend/target/release/build/ring-693946ddc750ed2d/out -L native=/home/thunderwang/git/databend/target/release/build/lz4-sys-f6b2205220dab049/out -L native=/home/thunderwang/git/databend/target/release/build/tikv-jemalloc-sys-b40a5802bf6a9b34/out/build/lib  (signal: 9, SIGKILL: kill)
warning: build failed, waiting for other jobs to finish...
error: build failed
`
please help me, thanks.

**Comment by Xuanwo** [comment, marked as answer]
The compiler has been kill by OS. It's highly possible that your memory is too small and the compiler OOM. Please consider allocate more virtual memory.

**Reply by chinawch007** [reply]
I increase the memory limit of the vm，it works.
Thank you for your help！

**Comment by PsiACE** [comment]
one of the possible solutions is to increase your swap space

**Reply by chinawch007** [reply]
Thank you for your answer. The problem was solved.
